src/enact/utils.py

```diff
@@ -30,9 +31,31 @@ def __init__(self, user_function):
 
   def __get__(self, instance: Any, unused_owner: Optional[type] = None):
     """Cached property based on the digest of the resource."""
-    digest = enact.resource_digest(instance)
+    digest = resource_digests.resource_digest(instance)
     result = self._cache.get(digest, None)
     if result is None:
       result = self._user_function(instance)
       self._cache = {digest: result}
     return result
+
+
+def walk_resource_dict(
+    value: interfaces.ResourceDictValue,
+    include_self: bool = True) -> Iterable[interfaces.ResourceDict]:
+  """Recursively yields all ResourceDict occurrences in the dictionary tree.
+
+  Args:
+    value: The value to walk.
+    include_self: Whether to also yield 'value' if it is a resource dict.
+
+  Yields:
+    All instances of ResourceDict in the subtree defined by 'value'.
```

Comment:
nit: Worth describing the order of traversal? The test case looks like it assumes depth first.

Reply:
Done.